stdlib/public/TensorFlow/TensorShape.swift

```diff
@@ -75,65 +75,76 @@ public struct TensorShape : ExpressibleByArrayLiteral {
 public extension TensorShape {
   /// The rank of the shape (i.e. the number of dimensions).
   @inlinable
-  var count: Int32 {
+  var count: Int {
     @inline(__always)
     get {
-      return Int32(dimensions.count)
+      return dimensions.count
     }
   }
 
   @inlinable
-  var indices: Range<Int32> {
+  var indices: Range<Int> {
     @inline(__always)
     get {
-      return Int32(dimensions.indices.lowerBound)
-        ..< Int32(dimensions.indices.upperBound)
+      return dimensions.indices.lowerBound
+        ..< dimensions.indices.upperBound
     }
   }
 
   @inlinable
-  var startIndex: Int32 {
+  var startIndex: Int {
     @inline(__always)
     get {
-      return Int32(dimensions.startIndex)
+      return dimensions.startIndex
     }
   }
 
   @inlinable
-  var endIndex: Int32 {
+  var endIndex: Int {
     @inline(__always)
     get {
-      return Int32(dimensions.endIndex)
+      return dimensions.endIndex
     }
   }
 
   /// Access the size of the i-th dimension.
   /// - Parameter index: The index of a dimension.
   @inlinable
-  subscript(index: Int32) -> Int32 {
+  subscript(index: Int) -> Int {
     @inline(__always)
     _read {
-      yield dimensions[Int(index)]
+      yield dimensions[index]
     }
     @inline(__always)
     _modify {
-      yield &dimensions[Int(index)]
+      yield &dimensions[index]
     }
   }
 
   /// Access the size of the i-th dimension.
   /// - Parameter index: The index of a dimension.
   @inlinable
-  subscript(bounds: Range<Int32>) -> TensorShape {
+  subscript(index: Int32) -> Int {
+    @inline(__always)
+    _read {
+      yield self[Int(index)]
+    }
+    @inline(__always)
+    _modify {
+      yield &self[Int(index)]
+    }
+  }
+
+  /// Access the size of the i-th dimension.
+  /// - Parameter index: The index of a dimension.
+  @inlinable
+  subscript(bounds: Range<Int>) -> TensorShape {
     @inline(__always)
     get {
-      return TensorShape(
-        Array(dimensions[Int(bounds.lowerBound)..<Int(bounds.upperBound)])
-      )
+      return TensorShape(Array(dimensions[bounds]))
```

Comment:
I thought `TensorShape` has a collection initializer, so i don't think `Array(...)` is necessary.

Reply:
`TensorShape` does not yet have a collection initializer. I will add one.